data-prepper-plugins/common/src/test/java/org/opensearch/dataprepper/plugins/s3keyindex/S3ObjectIndexUtilityTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.s3keyindex;
+
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import org.junit.jupiter.api.Test;
+
+class S3ObjectIndexUtilityTest {
+
+    @Test
+    void testObjectDateTimePatterns_not_equal() throws IllegalArgumentException {
+
+        String expectedIndex = S3ObjectIndexUtility.getObjectNameWithDateTimeId("events-%{yyyy-MM-dd}");
+        String actualIndex = S3ObjectIndexUtility.getObjectNameWithDateTimeId("events-%{yyyy-MM-dd}");
+        assertFalse(actualIndex.contains(expectedIndex));
+    }
+    
+    @Test
+    void testgetObjectPathPrefix_not_equal() throws IllegalArgumentException {
+
+        String expectedIndex = S3ObjectIndexUtility.getObjectPathPrefix("events-%{yyyy}");
+        String actualIndex = S3ObjectIndexUtility.getObjectPathPrefix("events-%{yyyy}");
+        assertTrue(actualIndex.contains(expectedIndex));
+    }
+
+    @Test
+    void testObjectTimePattern_Exceptional_time_TooGranular() throws IllegalArgumentException {
+        assertThrows(IllegalArgumentException.class, () -> {
+            S3ObjectIndexUtility.getDatePatternFormatter("events-%{yyyy-AA-dd}");
+        });
+    }
+
+    @Test
+    void testObjectTimePatterns_equal() throws IllegalArgumentException {
+
+        DateTimeFormatter expectedIndex = S3ObjectIndexUtility.getDatePatternFormatter("events-%{yyyy-MM-dd}");
+        DateTimeFormatter actualIndex = S3ObjectIndexUtility.getDatePatternFormatter("events-%{yyyy-MM-dd}");
```

Comment:
actualIndex -> actualDateTimeFormatter

Reply:
Also it is recommended to use a slightly different indexAlias as long as the same datetime pattern is contained.

Reply:
Resolved.